core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -227,13 +235,16 @@ private Decision canBeAllocatedToAtLeastOneNode(ShardRouting shard, RoutingAlloc
             // if we can't allocate it on a node, ignore it, for example, this handles
             // cases for only allocating a replica after a primary
             Decision decision = allocation.deciders().canAllocate(shard, node, allocation);
```

Comment:
Same comment here about `allocation.debugDecision()` and the flags that should be present when `explain` is true so decisions aren't short-circuited